tests/discovery_test.js

```diff
@@ -4,13 +4,341 @@ import chai, { expect } from "chai";
+      beforeEach(() => {
+        sandbox.stub(root, "fetch").returns(fakeServerResponse(200, {
+          data: {url: userStorageURL}
+        }, {}));
+      });
+
+      it("should return the existing URL", () => {
+        return registerUserURL("userID", centralRepositoryURL, headers, userStorageURL)
+        .should.become(userStorageURL);
+      });
+    });
+
+    describe("when server returns a 5xx", () => {
+      userStorageURL = "https://my-kinto-instance.com/v1";
+      const defaultServer = "https://default-kinto-instance.com/v1"
+
+
+      it("should return an error in case of 501 ", () => {
+        sandbox.stub(root, "fetch").returns(fakeServerResponse(501, {
+          data: {url: userStorageURL}
+          }, {} ));
+        return registerUserURL("userID", centralRepositoryURL, headers, userStorageURL)
+        .should.be.rejectedWith(Error);
+      })
+
+      it("should return an error in case of 500", () => {
+        sandbox.stub(root, "fetch").returns(fakeServerResponse(500, {
+          data: {url: userStorageURL}
+          }, {} ));
+        return registerUserURL("userID", centralRepositoryURL, headers, userStorageURL)
+        .should.be.rejectedWith(Error);
+      })
+
+      it("should return an error in case of 503", () => {
+        sandbox.stub(root, "fetch").returns(fakeServerResponse(503, {
+          data: {url: userStorageURL}
+          }, {} ));
+        return registerUserURL("userID", centralRepositoryURL, headers, userStorageURL)
+        .should.be.rejectedWith(Error);
+      })
+     });
+
+
+         describe("does not fit in any case", () => {
+           const defaultServer = "https://default-kinto-instance.com/v1"
+           beforeEach(() => {
+             sandbox.stub(root, "fetch").returns(fakeServerResponse(550, { //any response apart from the if block cases
+               data: {url: userStorageURL}
+             }, {}));
+           });
+
+           it("should reject promise with error message", () => {
+             return registerUserURL("userID", centralRepositoryURL, headers, userStorageURL)
+             .should.be.rejectedWith(Error);
+           });
+         });
+
+    describe("if headers are empty", () => {
+      userStorageURL = "https://my-kinto-instance.com/v1";
+      var headers = {};
+      it("should return an error message", () => {
+        registerUserURL("userID", centralRepositoryURL, headers, userStorageURL)
+        .should.become(Error);
+      });
+    });
+
+    describe("When there is already a cached value", () => {
+      var key = 'kinto:server-url:' + 'userID';
+      const defaultServer = "https://default-kinto-instance.com/v1"
+      beforeEach(() => {
+        localStorage.setItem(key, userStorageURL);
+      });
+    afterEach(() => {
+      localStorage.removeItem(key);
+    });
+
+      it("should return the cachedURL", () => {
+        return registerUserURL("userID", centralRepositoryURL, headers, userStorageURL)
```

Comment:
I think when we register a value we want to do it even if we have a cached value.